Edgex 315 create a standalone edgex ekuiper snap

MonicaisHer opened the pull request:
<!-- Expected Commit Message Description (imported automatically by GitHub) -->
<!-- Must conform to [conventional commits guidelines](https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md) -->
<!-- Expected Commit message must contain Closes/Fixes #IssueNumber statement when there is a related issue -->

<!-- Add additional detailed description of need for change if no related issue -->

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions
<!-- How can the reviewers test your change? -->

## New Dependency Instructions (If applicable)
<!-- Please follow [vetting instructions](https://wiki.edgexfoundry.org/display/FA/Vetting+Process+for+3rd+Party+Dependencies) and place results here -->

farshidtz commented on snap/snapcraft.yaml:
```diff
@@ -67,9 +67,16 @@ plugs:
+    interface: content
+    source:
+      read: 
+        - $SNAP/etc/sources
+        - $SNAP/etc/connections
```
Change to:
```suggestion
        - $SNAP_DATA/kuiper/etc/sources
        - $SNAP_DATA/kuiper/etc/connections
```
Then connect
```
snap connect edgex-ekuiper:kuiper-config edgexfoundry:kuiper-config
```
```
$ sudo snap run --shell edgex-ekuiper.kuiper
# sudo ls /var/snap/edgex-ekuiper/current/etc/sources -l
# ls /var/snap/edgex-ekuiper/current/etc/sources -l
total 12
-rw-r--r-- 1 root root 8137 Feb 11 14:42 edgex.json
-rw-r--r-- 1 root root  440 Feb 11 15:37 edgex.yaml <-- note the modification time
```
Disconnect and compare the time:
```
# ls /var/snap/edgex-ekuiper/current/etc/sources -l
total 12
-rw-r--r-- 1 root root 8137 Feb 11 14:49 edgex.json
-rw-r--r-- 1 root root 1221 Feb 11 14:49 edgex.yaml
```

MonicaisHer replied:
Thanks. How to validate the content of modified `edgex.yaml` is correct inside the snap?

farshidtz replied:
You mean other than manually checking with cat or vi?

MonicaisHer replied:
Please ignore my comment. I was in the wrong place.

MonicaisHer committed:
feat(snap): add secretstore tokens for edgex-ekuiper

The Vault token is required for the standalone deployment of eKuiper with snap, 
where a script queries the Redis credentials after authenticating with Vault.
Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>

MonicaisHer commented:
This PR will be ready after [edgex-ekuiper PR ](https://github.com/MonicaisHer/edgex-ekuiper-snap/pull/2)
